torchgen/model.py

```diff
@@ -728,6 +756,7 @@ def has_composite_kernel(self) -> bool:
         return (
             self.has_composite_implicit_autograd_kernel
             or self.has_composite_explicit_autograd_kernel
+            or len(self.composite) > 0
```

Comment:
No action needed, but this semantics of this function are kind of confusing. Today at least, it really means "has an explicitly mentioned `CompositeImplicitAutograd/CompositeExplicitAutograd` entry in `native_functions.yaml`.

It seems reasonable to have it mean "does this op get a composite kernel registered to the dispatcher", although it doesn't really do that today, since the codegen will generate `CompositeExplicitAutograd` kernels for all `functional`/`inplace` structured operators.

Reply:
> Today at least, it really means "has an explicitly mentioned `CompositeImplicitAutograd/CompositeExplicitAutograd`...

That's only true for `CompositeExplicitAutograd`, right!? If the kernel has no explicit dispatch, `has_composite_implicit_autograd_kernel` also gets set.

That being said, I think this line isn't really needed anymore (I thought I needed it, before). I think it's better to remove it. Let me know if you think otherwise.

https://github.com/pytorch/pytorch/blob/7a09a4ebc669e2264ccba69a40e9d6757a214655/torchgen/model.py#L589-L592